### DISCUSSION TITLE
chunk type is data is different from photoshop saved png

### DISCUSSION THREAD

**Discussion by srxqds** [SixLabors/ImageSharp#2354] [Q&A] [answered]
I use ImageSharp save png, but it is different from saved from photoshop.
I found the metadata is different, show as below(first is imagesharp saved, second is photoshop)
```
Data length: 9 798 bytes
Type: IDAT
Name: Image data
Critical (0)
Public (0)
Reserved (0)
Unsafe to copy (0)
CRC-32: 55FEF5F4
```

```
Data length: 8 192 bytes
Type: IDAT
Name: Image data
Critical (0)
Public (0)
Reserved (0)
Unsafe to copy (0)
CRC-32: C430AAE1

Data length: 947 bytes
Type: IDAT
Name: Image data
Critical (0)
Public (0)
Reserved (0)
Unsafe to copy (0)
CRC-32: 98925A7E
```

how can we set to keep them all the same.

**Comment by JimBobSquarePants** [comment, marked as answer]
The IDAT information depends on several factors. 

1. The Png color type used - Rgb24, Rgba32 etc.
2. The compression level used. 1-9
3. Whether the data is interlaced.
4. What filtering method is used to rearrange the scanlines before compression (Paeth, Average, etc.)
4. What approach is used to split the data across multiple chunks. 
5. What ZLib implementation is used to compress the data.

You're extremely unlikely to guarantee identical IDAT chunks unless both encoder implementations are using the exact same properties for all those items.

So, the question begs... Why? Why would you be attempting to do something like this?

**Reply by srxqds** [reply]
thank you for your reply, maybe the diff is caused by the 5. approach split data, other config seems same.
how can I set the approach split data?

**Reply by srxqds** [reply]
these metadata are all the same :
```
Width: 82 pixels
Height: 84 pixels
Bit depth: 8 bits per channel
Color type: RGBA (6)
Compression method: DEFLATE (0)
Filter method: Adaptive (0)
Interlace method: None (0)
```

**Reply by JimBobSquarePants** [reply]
More likely Zlib implementation

**Comment by srxqds** [comment]
we need to keep ImageSharp to save png file same with exported from photoshop

**Reply by JimBobSquarePants** [reply]
Why should the encoded data match byte for byte? Like I said different encoders will produce different results. 

If you encoded the same image using System.Drawing, SkiaSharp, and ImageSharp the encoded bytes would not only be different to each other but different to Photoshop.

**Reply by JimBobSquarePants** [reply]
Remember this doesn’t affect the pixel data when decoding is repeated. The encoder is lossless by default
